plugins/package-managers/gradle-inspector/src/main/kotlin/GradleDependencyHandler.kt

```diff
@@ -83,18 +83,18 @@ internal class GradleDependencyHandler : DependencyHandler<OrtDependency> {
             }
         }
 
-        val isMetadataOnly = dependency.extension == "pom" || isSpringMetadataProject
+        val hasNoArtifacts = dependency.pomFile == null || isSpringMetadataProject
 
         val binaryArtifact = when {
-            isMetadataOnly -> RemoteArtifact.EMPTY
+            hasNoArtifacts -> RemoteArtifact.EMPTY
             else -> with(dependency) {
-                createRemoteArtifact(pomFile, classifier, extension.takeUnless { it == "bundle" })
+                createRemoteArtifact(checkNotNull(pomFile), classifier, extension.takeUnless { it == "bundle" })
```

Comment:
Commit message: Why do you assume that `pomFile` is not `null` here? `hasNoArtifacts` could also be `true` because `isSpringMetadataProject` is `true`.

Reply:
Right, I was trying to combine too many things here. Fixed.